public/app/features/alerting/unified/AmRoutes.tsx

```diff
@@ -15,10 +16,17 @@ const AmRoutes: FC = () => {
   const amConfigs = useUnifiedAlertingSelector((state) => state.amConfigs);
 
   useEffect(() => {
-    dispatch(fetchAlertManagerConfigAction(alertManagerSourceName));
+    if (alertManagerSourceName) {
+      dispatch(fetchAlertManagerConfigAction(alertManagerSourceName));
+    }
   }, [alertManagerSourceName, dispatch]);
 
-  const { result, loading, error } = amConfigs[alertManagerSourceName] || initialAsyncRequestState;
+  const { result, loading, error } =
+    (alertManagerSourceName && amConfigs[alertManagerSourceName]) || initialAsyncRequestState;
+
+  if (!alertManagerSourceName) {
+    return <Redirect to="/alerting/routes" />;
```

Comment:
I investigated to be certain.  `config.appSubUrl` is set  as `basename` [on the history object](https://github.com/grafana/grafana/blob/master/packages/grafana-runtime/src/services/LocationService.ts#L36).
This means that URLs:
* Do not have to be prefixed when interacting with react router, it is handled automatically
* Do not have to be prefixed when interacting with `LocationService`, handled automatically
* Do have to be prefixed when settin an anchor element's href directly, eg when using `<a />` or `<LinkButton />`. If you do not prefix, it will work anyway if already inside grafana, but such links will not work if copied, opened in new page or bookmarked

I went over the code and made sure that it's correct in all cases for unified alering. Also updated ot use `locationUtils` instead of `config` directly where appropriate

Reply:
made all anchor hrefs relative, base url will be handled automatically by browser because of `<base/>`